internal/codegenerator/condition_table.go

```diff
@@ -0,0 +1,354 @@
+				operator:       "<=",
+				roperand:       `{{.Target}}`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} must be <= {{.Target}}",
+			},
+		},
+	},
+	"min": {
+		Name: "min",
+		ConditionByType: map[string]ConditionTable{
+			"string": {
+				loperand:       "len({{.Name}})",
+				operator:       ">=",
+				roperand:       `{{.Target}}`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} length must be >= {{.Target}}",
+			},
+			"[]string": {
+				loperand:       "len({{.Name}})",
+				operator:       ">=",
+				roperand:       `{{.Target}}`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} must have at least {{.Target}} elements",
+			},
+		},
+	},
+	"max": {
+		Name: "max",
+		ConditionByType: map[string]ConditionTable{
+			"string": {
+				loperand:       "len({{.Name}})",
+				operator:       "<=",
+				roperand:       `{{.Target}}`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} length must be <= {{.Target}}",
+			},
+			"[]string": {
+				loperand:       "len({{.Name}})",
+				operator:       "<=",
+				roperand:       `{{.Target}}`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} must have at most {{.Target}} elements",
+			},
+		},
+	},
+	"eq_ignore_case": {
+		Name: "eq_ignore_case",
+		ConditionByType: map[string]ConditionTable{
+			"string": {
+				loperand:       "types.ToLower({{.Name}})",
+				operator:       "==",
+				roperand:       `"{{.Target}}"`,
+				concatOperator: "",
+				normalizeFunc:  strings.ToLower,
+				errorMessage:   "{{.Name}} must be equal to '{{.Target}}'",
+			},
+		},
+	},
+	"len": {
+		Name: "len",
+		ConditionByType: map[string]ConditionTable{
+			"string": {
+				loperand:       "len({{.Name}})",
+				operator:       "==",
+				roperand:       `{{.Target}}`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} length must be {{.Target}}",
+			},
+			"[]string": {
+				loperand:       "len({{.Name}})",
+				operator:       "==",
+				roperand:       `{{.Target}}`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} must have exactly {{.Target}} elements",
+			},
+		},
+	},
+	"neq": {
+		Name: "neq",
+		ConditionByType: map[string]ConditionTable{
+			"string": {
+				loperand:       "{{.Name}}",
+				operator:       "!=",
+				roperand:       `"{{.Target}}"`,
+				concatOperator: "",
+				normalizeFunc:  nil,
+				errorMessage:   "{{.Name}} must not be equal to '{{.Target}}'",
+			},
+		},
+	},
+	"neq_ignore_case": {
+		Name: "neq_ignore_case",
+		ConditionByType: map[string]ConditionTable{
+			"string": {
+				loperand:       "types.ToLower({{.Name}})",
+				operator:       "!=",
+				roperand:       `"{{.Target}}"`,
+				concatOperator: "",
+				normalizeFunc:  strings.ToLower,
+				errorMessage:   "{{.Name}} must not be equal to '{{.Target}}'",
```

Comment:
```suggestion
				operation:       "types.ToLower({{.Name}}) != {{.Target}}",
				concatOperator: "",
				normalizeFunc:  strings.ToLower,
				errorMessage:   "{{.Name}} must not be equal to '{{.Target}}'",
```
Não poderia juntar em um único?

Inclusive o `lower() == lower()` poderia ser substituido por [EqualFold](https://pkg.go.dev/strings#EqualFold)

Reply:
Juntei loperand, operator e roperator em um campo só. Fiquei BEM melhor. Valeu @fabriciogpp